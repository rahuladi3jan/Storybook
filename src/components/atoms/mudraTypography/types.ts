import React from "react";
import colors from "../../../styles/styledComponents/colors";

export type Sizes =
  | 10
  | 12
  | 14
  | 16
  | 18
  | 20
  | 24
  | 28
  | 32
  | 36
  | 40
  | 48
  | 56
  | 64
  | 72
  | 80
  | 88
  | 96;

export type Weight = "regular" | "bold" | "extraBold";

type ColorKeys<T> = keyof T;
type ColorShades = {
  [K in ColorKeys<typeof colors>]: keyof typeof colors[K];
};
type GenerateColorVariants<T extends Record<string, number>> = {
  [K in keyof T & string as `${string & K}${T[K]}`]: `${string & K}${T[K]}`;
};
type ColorVariants = GenerateColorVariants<ColorShades>;
export type ColorVariant = ColorVariants[keyof ColorVariants];

export interface MudraTypographyProps
  extends React.HTMLAttributes<HTMLDivElement> {
  weight?: Weight;
  size?: Sizes;
  textColor?: ColorVariant;
  underline?: boolean;
  capitalize?: boolean;
  ellipsis?: boolean;
  lineThrough?: boolean;
}
