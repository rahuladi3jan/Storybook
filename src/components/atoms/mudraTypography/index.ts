export { default as MudraTypography } from "./mudraTypography";
