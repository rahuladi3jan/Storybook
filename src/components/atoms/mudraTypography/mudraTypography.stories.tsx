import React from "react";
import { Meta } from "@storybook/react";

import MudraTypography from "./mudraTypography";
import { Sizes } from "./types";

export default {
  component: MudraTypography,
  title: "Atoms/Mudra Typography",
  parameters: {
    componentSubtitle: "Storybook Mudra Typography",
    docs: {
      description: {
        component:
          "MudraTypography enables the incorporation of every conceivable text variant from your Figma design using props exclusively. Each prop is equipped with value suggestions, streamlining the customization process.",
      },
    },
  },
} as Meta;

export const ATextNunito = () => {
  const sizes: Sizes[] = [
    10, 12, 14, 16, 18, 20, 24, 28, 32, 36, 40, 48, 56, 64, 72, 80, 88, 96,
  ];
  return (
    <div>
      <MudraTypography
        size={40}
        textColor="storybookOrange100"
        underline
        weight="extraBold"
      >
        Nunito Regular Typography
      </MudraTypography>
      {/* regular typography with all text size */}
      {sizes.map((size) => {
        return (
          <MudraTypography key={size} size={size}>
            Nunito - regular - {size}
          </MudraTypography>
        );
      })}

      <MudraTypography
        size={40}
        textColor="storybookOrange100"
        underline
        weight="extraBold"
      >
        Nunito Bold Typography
      </MudraTypography>
      {/* bold typography with all text size */}
      {sizes.map((size) => {
        return (
          <MudraTypography key={size} size={size} weight="bold">
            Nunito - bold - {size}
          </MudraTypography>
        );
      })}

      <MudraTypography
        size={40}
        textColor="storybookOrange100"
        underline
        weight="extraBold"
      >
        Nunito Extra Bold Typography
      </MudraTypography>
      {/* extraBold typography with all text size */}
      {sizes.map((size) => {
        return (
          <MudraTypography key={size} size={size} weight="extraBold">
            Nunito - extraBold - {size}
          </MudraTypography>
        );
      })}
    </div>
  );
};

export const BVariants = () => {
  return (
    <div>
      <MudraTypography size={20} textColor="storybookBlue120" weight="extraBold">
        Storybook blue 120 - text size 20 - extra bold
      </MudraTypography>
      <MudraTypography size={24} textColor="storybookOrange140" weight="bold">
        Storybook orange 140 - text size 24 - bold
      </MudraTypography>
      <MudraTypography size={28} textColor="royalBlue60">
        Royal blue 60 - text size 28 - regular
      </MudraTypography>
    </div>
  );
};

export const CCapitalize = () => (
  <MudraTypography capitalize>Capitalize</MudraTypography>
);

export const DUnderline = () => (
  <MudraTypography underline>Underline</MudraTypography>
);

export const EEllipsis = () => (
  <MudraTypography ellipsis>
    Ellipsis Text Ellipsis TextEllipsis Text Ellipsis Text Ellipsis Text
    Ellipsis Text Ellipsis Text Ellipsis Text Ellipsis Text Ellipsis Text
    Ellipsis Text Ellipsis Text Ellipsis Text
  </MudraTypography>
);

export const FLineThrough = () => (
  <MudraTypography lineThrough>Line Through</MudraTypography>
);

// Define the order of your stories
ATextNunito.storyName = "TextNunito";
BVariants.storyName = "Variants";
CCapitalize.storyName = "Capitalize";
DUnderline.storyName = "Underline";
EEllipsis.storyName = "Ellipsis";
FLineThrough.storyName = "LineThrough";
