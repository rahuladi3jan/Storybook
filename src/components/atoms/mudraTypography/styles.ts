import styled, { css } from "styled-components";
import { lightTheme } from "../../../styles/styledComponents/theme";
import { MudraTypographyProps } from "./types";

const { colors, lineHeight, sizes } = lightTheme;

const getColorValue = (textColor): string => {
  const [colorName, colorCode] = textColor
    .match(/([a-zA-Z]+)([0-9]+)/)!
    .slice(1);
  return colors[colorName][parseInt(colorCode)];
};

export const Typography = styled.span<MudraTypographyProps>`
  display: block;
  font-weight: 400;
  margin: 0;
  ${({
    textColor,
    size,
    weight,
    underline,
    capitalize,
    ellipsis,
    lineThrough,
  }) => css`
    color: ${getColorValue(textColor)};
    font-size: ${sizes[size]};
    line-height: ${lineHeight[size]};
    ${weight === "bold" &&
    css`
      font-weight: 700;
    `}
    ${weight === "extraBold" &&
    css`
      font-weight: 800;
    `}
    ${ellipsis &&
    css`
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    `}
    ${underline &&
    css`
      text-decoration: underline;
    `}
    ${capitalize &&
    css`
      text-transform: uppercase;
    `}
    ${lineThrough &&
    css`
      text-decoration: line-through;
    `}
  `}
`;
