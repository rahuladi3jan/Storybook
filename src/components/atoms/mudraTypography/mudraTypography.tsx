import React from "react";

import { Typography } from "./styles";
import { MudraTypographyProps } from "./types";

function MudraTypography(props: MudraTypographyProps) {
  const { children } = props;

  return <Typography {...props}>{children}</Typography>;
}

MudraTypography.defaultProps = {
  weight: "regular",
  size: 16,
  textColor: "neutral90",
  capitalize: false,
  underline: false,
  ellipsis: false,
  lineThrough: false,
};

export default MudraTypography;
