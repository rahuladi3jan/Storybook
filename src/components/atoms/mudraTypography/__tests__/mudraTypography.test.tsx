import renderer from "../../../../utils/testRenderer";

import * as tests from "../mudraTypography.stories";

renderer(tests);
