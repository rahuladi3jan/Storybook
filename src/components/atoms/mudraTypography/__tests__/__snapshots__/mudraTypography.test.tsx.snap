// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Mudra Typography ATextNunito 1`] = `
<div>
  <span
    className="sc-bdVaJa iTFIoM"
    size={40}
  >
    Nunito Regular Typography
  </span>
  <span
    className="sc-bdVaJa dAPAgl"
    size={10}
  >
    Nunito - regular -
    10
  </span>
  <span
    className="sc-bdVaJa MUnUL"
    size={12}
  >
    Nunito - regular -
    12
  </span>
  <span
    className="sc-bdVaJa ekdSXV"
    size={14}
  >
    Nunito - regular -
    14
  </span>
  <span
    className="sc-bdVaJa lkLaLf"
    size={16}
  >
    Nunito - regular -
    16
  </span>
  <span
    className="sc-bdVaJa bkXaww"
    size={18}
  >
    Nunito - regular -
    18
  </span>
  <span
    className="sc-bdVaJa WGFCP"
    size={20}
  >
    Nunito - regular -
    20
  </span>
  <span
    className="sc-bdVaJa gOwJSr"
    size={24}
  >
    Nunito - regular -
    24
  </span>
  <span
    className="sc-bdVaJa dgRblo"
    size={28}
  >
    Nunito - regular -
    28
  </span>
  <span
    className="sc-bdVaJa jJgsKw"
    size={32}
  >
    Nunito - regular -
    32
  </span>
  <span
    className="sc-bdVaJa gxXNrn"
    size={36}
  >
    Nunito - regular -
    36
  </span>
  <span
    className="sc-bdVaJa cbBsLN"
    size={40}
  >
    Nunito - regular -
    40
  </span>
  <span
    className="sc-bdVaJa ekOcWj"
    size={48}
  >
    Nunito - regular -
    48
  </span>
  <span
    className="sc-bdVaJa kTPmYn"
    size={56}
  >
    Nunito - regular -
    56
  </span>
  <span
    className="sc-bdVaJa emnqLz"
    size={64}
  >
    Nunito - regular -
    64
  </span>
  <span
    className="sc-bdVaJa icPFOU"
    size={72}
  >
    Nunito - regular -
    72
  </span>
  <span
    className="sc-bdVaJa kAWBks"
    size={80}
  >
    Nunito - regular -
    80
  </span>
  <span
    className="sc-bdVaJa IpQNv"
    size={88}
  >
    Nunito - regular -
    88
  </span>
  <span
    className="sc-bdVaJa kQgRMX"
    size={96}
  >
    Nunito - regular -
    96
  </span>
  <span
    className="sc-bdVaJa iTFIoM"
    size={40}
  >
    Nunito Bold Typography
  </span>
  <span
    className="sc-bdVaJa jSUkpy"
    size={10}
  >
    Nunito - bold -
    10
  </span>
  <span
    className="sc-bdVaJa hpFkdd"
    size={12}
  >
    Nunito - bold -
    12
  </span>
  <span
    className="sc-bdVaJa bIwFdK"
    size={14}
  >
    Nunito - bold -
    14
  </span>
  <span
    className="sc-bdVaJa hLJofu"
    size={16}
  >
    Nunito - bold -
    16
  </span>
  <span
    className="sc-bdVaJa dNgXdJ"
    size={18}
  >
    Nunito - bold -
    18
  </span>
  <span
    className="sc-bdVaJa beWCzw"
    size={20}
  >
    Nunito - bold -
    20
  </span>
  <span
    className="sc-bdVaJa klODJR"
    size={24}
  >
    Nunito - bold -
    24
  </span>
  <span
    className="sc-bdVaJa goZfFW"
    size={28}
  >
    Nunito - bold -
    28
  </span>
  <span
    className="sc-bdVaJa eGMjoJ"
    size={32}
  >
    Nunito - bold -
    32
  </span>
  <span
    className="sc-bdVaJa epEscu"
    size={36}
  >
    Nunito - bold -
    36
  </span>
  <span
    className="sc-bdVaJa gXKYGC"
    size={40}
  >
    Nunito - bold -
    40
  </span>
  <span
    className="sc-bdVaJa jOwAtn"
    size={48}
  >
    Nunito - bold -
    48
  </span>
  <span
    className="sc-bdVaJa iBlKxK"
    size={56}
  >
    Nunito - bold -
    56
  </span>
  <span
    className="sc-bdVaJa fHrbvj"
    size={64}
  >
    Nunito - bold -
    64
  </span>
  <span
    className="sc-bdVaJa jRNLhA"
    size={72}
  >
    Nunito - bold -
    72
  </span>
  <span
    className="sc-bdVaJa fpzFgx"
    size={80}
  >
    Nunito - bold -
    80
  </span>
  <span
    className="sc-bdVaJa hGFvW"
    size={88}
  >
    Nunito - bold -
    88
  </span>
  <span
    className="sc-bdVaJa fYJLlw"
    size={96}
  >
    Nunito - bold -
    96
  </span>
  <span
    className="sc-bdVaJa iTFIoM"
    size={40}
  >
    Nunito Extra Bold Typography
  </span>
  <span
    className="sc-bdVaJa gpAWlT"
    size={10}
  >
    Nunito - extraBold -
    10
  </span>
  <span
    className="sc-bdVaJa eyVLXm"
    size={12}
  >
    Nunito - extraBold -
    12
  </span>
  <span
    className="sc-bdVaJa kAIFnX"
    size={14}
  >
    Nunito - extraBold -
    14
  </span>
  <span
    className="sc-bdVaJa cOowfJ"
    size={16}
  >
    Nunito - extraBold -
    16
  </span>
  <span
    className="sc-bdVaJa feqjFq"
    size={18}
  >
    Nunito - extraBold -
    18
  </span>
  <span
    className="sc-bdVaJa QVdDP"
    size={20}
  >
    Nunito - extraBold -
    20
  </span>
  <span
    className="sc-bdVaJa hcPwTc"
    size={24}
  >
    Nunito - extraBold -
    24
  </span>
  <span
    className="sc-bdVaJa xnxdA"
    size={28}
  >
    Nunito - extraBold -
    28
  </span>
  <span
    className="sc-bdVaJa qnrPv"
    size={32}
  >
    Nunito - extraBold -
    32
  </span>
  <span
    className="sc-bdVaJa kfwsZl"
    size={36}
  >
    Nunito - extraBold -
    36
  </span>
  <span
    className="sc-bdVaJa ffeVbv"
    size={40}
  >
    Nunito - extraBold -
    40
  </span>
  <span
    className="sc-bdVaJa eJwXye"
    size={48}
  >
    Nunito - extraBold -
    48
  </span>
  <span
    className="sc-bdVaJa ijmOlP"
    size={56}
  >
    Nunito - extraBold -
    56
  </span>
  <span
    className="sc-bdVaJa izoklx"
    size={64}
  >
    Nunito - extraBold -
    64
  </span>
  <span
    className="sc-bdVaJa gqcxEe"
    size={72}
  >
    Nunito - extraBold -
    72
  </span>
  <span
    className="sc-bdVaJa jFrtRK"
    size={80}
  >
    Nunito - extraBold -
    80
  </span>
  <span
    className="sc-bdVaJa gowviT"
    size={88}
  >
    Nunito - extraBold -
    88
  </span>
  <span
    className="sc-bdVaJa jYupZb"
    size={96}
  >
    Nunito - extraBold -
    96
  </span>
</div>
`;

exports[`Atoms/Mudra Typography BVariants 1`] = `
<div>
  <span
    className="sc-bdVaJa fUiqOY"
    size={20}
  >
    Storybook blue 120 - text size 20 - extra bold
  </span>
  <span
    className="sc-bdVaJa kkyKFo"
    size={24}
  >
    Storybook orange 140 - text size 24 - bold
  </span>
  <span
    className="sc-bdVaJa gsCvsR"
    size={28}
  >
    Royal blue 60 - text size 28 - regular
  </span>
</div>
`;

exports[`Atoms/Mudra Typography CCapitalize 1`] = `
<span
  className="sc-bdVaJa gmAawY"
  size={16}
>
  Capitalize
</span>
`;

exports[`Atoms/Mudra Typography DUnderline 1`] = `
<span
  className="sc-bdVaJa bmJQJN"
  size={16}
>
  Underline
</span>
`;

exports[`Atoms/Mudra Typography EEllipsis 1`] = `
<span
  className="sc-bdVaJa fFiHal"
  size={16}
>
  Ellipsis Text Ellipsis TextEllipsis Text Ellipsis Text Ellipsis Text Ellipsis Text Ellipsis Text Ellipsis Text Ellipsis Text Ellipsis Text Ellipsis Text Ellipsis Text Ellipsis Text
</span>
`;

exports[`Atoms/Mudra Typography FLineThrough 1`] = `
<span
  className="sc-bdVaJa dPRcxI"
  size={16}
>
  Line Through
</span>
`;
